Dockerfile

```diff
@@ -49,18 +48,19 @@ RUN \
 # copy sources
 COPY . /home/jhipster/generator-jhipster
 
+# When https://github.com/docker/docker/pull/28499 will be merged and released,
+# remove the following line and add "--chown jhipster:jhipster" to the previous one
 RUN \
   # fix jhipster user permissions
-  chown -R jhipster:jhipster \
-    /home/jhipster \
-    /usr/lib/node_modules && \
+  chown -R jhipster:jhipster /home/jhipster/generator-jhipster
 
+RUN \
   # install jhipster
-  rm -Rf /home/jhipster/generator-jhipster/node_modules \
+  rm -rf /home/jhipster/generator-jhipster/node_modules \
     /home/jhipster/generator-jhipster/yarn.lock \
     /home/jhipster/generator-jhipster/yarn-error.log && \
   su -c "cd /home/jhipster/generator-jhipster && yarn install" jhipster && \
-  su -c "yarn global add file:/home/jhipster/generator-jhipster" jhipster && \
+  yarn global add file:/home/jhipster/generator-jhipster && \
```

Comment:
By not install this with jhipster user, generator-jhipster will be installed under root user:

With jhipster user:
```
jhipster@d0ab44b6078e:~/app$ ls -al /home/jhipster/.config/yarn/global/node_modules/generator-jhipster
ls: cannot access '/home/jhipster/.config/yarn/global/node_modules/generator-jhipster': No such file or directory
```

With root user:
```
root@d0ab44b6078e:~/.config# ls -al /root/.config/yarn/global/node_modules/generator-jhipster/
total 212
drwxr-xr-x   8 root root  4096 Mar  5 11:17 .
drwxr-xr-x 363 root root 12288 Mar  5 11:18 ..
-rw-r--r--   1 root root   268 Jan 15 21:07 .editorconfig
-rw-r--r--   1 root root    31 Feb 22 07:19 .eslintignore
-rw-r--r--   1 root root   478 Jan 15 21:07 .eslintrc.json
drwxr-xr-x   8 root root  4096 Mar  5 11:17 .git
-rw-r--r--   1 root root   434 Jan 15 21:07 .gitattributes
drwxr-xr-x   2 root root  4096 Mar  5 11:17 .github
-rw-r--r--   1 root root   113 Feb 22 07:33 .gitignore
-rw-r--r--   1 root root  2304 Mar  4 14:18 .travis.yml
-rw-r--r--   1 root root 13808 Jan 15 21:07 CONTRIBUTING.md
-rw-r--r--   1 root root  2222 Mar  5 11:12 Dockerfile
-rw-r--r--   1 root root 11454 Jan 15 21:07 LICENSE.txt
-rw-r--r--   1 root root  1991 Feb 22 07:19 README.md
drwxr-xr-x  19 root root  4096 Mar  5 11:17 generators
drwxr-xr-x 421 root root 16384 Mar  5 11:17 node_modules
-rw-r--r--   1 root root  1529 Mar  2 18:41 package.json
drwxr-xr-x   3 root root  4096 Mar  5 11:17 test
drwxr-xr-x   5 root root  4096 Mar  5 11:17 travis
-rw-r--r--   1 root root 96493 Mar  5 11:17 yarn.lock
```